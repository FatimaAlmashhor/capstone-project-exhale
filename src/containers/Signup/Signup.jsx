import React, { useState, useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { Link, useLocation } from 'react-router-dom';
import fire from '../../firebase';
import RegisterForm from '../../components/RegisterForm/RegisterForm';
import { LOGIN_ROUTE } from '../../routes';

const Signup = () => {
  const [isSignedIn, setisSignedIn] = useState(false);
  const location = useLocation();

  const { t } = useTranslation();
  useEffect(() => {
    fire.auth().onAuthStateChanged((user) => {
      setisSignedIn(!!user);
    });
  });

  return (
    <div>
      {isSignedIn ? null : (
        <>
          <div>
            <div className="my-2 text-center border-b">
              <div className="inline-block px-2 text-sm font-medium leading-none tracking-wide text-gray-600 transform translate-y-1/2 bg-white">
                {t('with email')}
              </div>
            </div>
            <RegisterForm />
            <div className="text-center ">
              <p className="mt-0 text-sm text-gray-500">
                {t('Already have an account')}
                <Link
                  className="text-blue-500 cursor-pointer "
                  to={{
                    pathname: LOGIN_ROUTE,
                    state: { modal: location.state.modal },
                  }}
                >
                  &nbsp;
                  {t('Login')}
                </Link>
              </p>
            </div>
          </div>
        </>
      )}
    </div>
  );
};

export default Signup;
