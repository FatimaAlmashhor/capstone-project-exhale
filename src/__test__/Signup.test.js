import React from 'react';
import renderer from 'react-test-renderer';
import Signup from '../containers/Signup';

it('renders correctly', () => {
    const fun = function (params) {
        // 
    }
    const foo= true;
  const tree = renderer.create(<Signup show={foo} onClick={fun}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
