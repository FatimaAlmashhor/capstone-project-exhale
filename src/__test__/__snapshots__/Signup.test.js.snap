// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="fixed inset-0 z-50 flex items-center justify-center overflow-x-hidden overflow-y-auto outline-none focus:outline-none"
    onClick={[Function]}
  >
    <div
      className="relative w-auto max-w-5xl mx-auto my-6"
      onClick={[Function]}
    >
      <div
        className="relative flex flex-col w-full bg-white border-0 rounded-lg shadow-lg outline-none focus:outline-none"
      >
        <div>
          <div
            className="flex items-start justify-between p-5 border-b border-gray-300 border-solid rounded-t"
          >
            <h3
              className="text-3xl font-semibold"
            >
              SING UP
            </h3>
            <button
              className="float-right p-1 ml-auto text-3xl font-semibold leading-none text-black bg-transparent border-0 outline-none opacity-5 focus:outline-none"
              onClick={[Function]}
              type="button"
            >
              <span
                className="block w-6 h-6 text-2xl text-black bg-transparent outline-none opacity-5 focus:outline-none"
              >
                Ã—
              </span>
            </button>
          </div>
          <div
            id="firebaseui_container"
          />
          <div
            className="relative flex-auto p-6"
          >
            <div>
              <div
                className="my-2 text-center border-b"
              >
                <div
                  className="inline-block px-2 text-sm font-medium leading-none tracking-wide text-gray-600 transform translate-y-1/2 bg-white"
                >
                  with email
                </div>
              </div>
              <div
                className="w-full max-w-md mx-auto"
              >
                <form
                  className=" pt-6 pb-8 mb-4 bg-white rounded w-full"
                  onSubmit={[Function]}
                >
                  <input
                    className="w-full p-3 leading-tight text-gray-700 border rounded  appearance-none focus:outline-none "
                    id="name"
                    name="name"
                    onChange={[Function]}
                    placeholder="yourNam"
                    type="text"
                  />
                  <input
                    className="mt-4 w-full p-3 leading-tight text-gray-700 border rounded  appearance-none focus:outline-none "
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="yourEmail"
                    type="text"
                  />
                  <input
                    className="mt-4 w-full p-3 leading-tight text-gray-700 border rounded  appearance-none focus:outline-none"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="yourPassword"
                    type="password"
                  />
                  <input
                    className="mt-4 w-full p-3 leading-tight text-gray-700 border rounded  appearance-none focus:outline-none"
                    id="confirmPassword"
                    name="confirmPassword"
                    onChange={[Function]}
                    placeholder="confirmYourPassword"
                    type="password"
                  />
                  <button
                    className="w-full px-4 py-3 mt-4 font-bold text-white bg-blue-800 rounded hover:bg-blue-700 focus:outline-none focus:shadow-outline"
                    type="submit"
                  >
                    SignUP
                  </button>
                  <div
                    className="text-lg text-red-500"
                  >
                    
                  </div>
                </form>
              </div>
              <div>
                <p
                  className="mt-5 text-sm text-gray-500"
                >
                  haveAccount
                  <a
                    className="text-blue-500 cursor-pointer "
                    href="/login"
                    onClick={[Function]}
                  >
                    Login
                  </a>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="fixed inset-0 z-40 bg-black opacity-25"
  />
</div>
`;
